### PR TITLE
LogReader: print number of missing rlogs

### LINKED ISSUE #32797: LogReader: count number of missing files
https://github.com/commaai/openpilot/blob/77a8094f55d35c6f51af72c2928ffe109c9609f9/tools/lib/logreader.py#L94-L105

### PR DESCRIPTION
**Pull request by StefanLekanic-SYRMIA**
#32799

### CODE REVIEW

**Comment by StefanLekanic-SYRMIA** [comment]
i wanted to do optimization to remove useless counting of qlogs in return statement, but we can leave that as it is, to minimize changes.

File: tools/lib/logreader.py [hunk 1]
```diff
@@ -90,15 +90,23 @@ class ReadMode(enum.StrEnum):
 def default_valid_file(fn: LogPath) -> bool:
   return fn is not None and file_exists(fn)
 
+def count_missing_rlogs(rlog_paths: LogPaths, valid_file: ValidFileCallable) -> int:
+  missing_count = 0
+  for rlog in rlog_paths:
+    if rlog is None or not valid_file(rlog):
+      missing_count += 1
+  return missing_count
```
Comment (sshane) [inline tools/lib/logreader.py]: cached the result so we don't take 2x longer

**Comment by sshane** [comment]
This PR is a slight speed up due to not checking qlogs upfront

**Comment by sshane** [comment]
thanks for getting this started!

**Pull request merged**